src/toolkits/activity_classification/activity_classifier.cpp

```diff
@@ -919,11 +919,21 @@ std::unique_ptr<model_spec> activity_classifier::init_model(
     std::seed_seq seed_seq{read_state<int>("random_seed")};
     random_engine = std::mt19937(seed_seq);
   }
-  result->add_channel_concat(
-      "features",
-      std::vector<std::string>(features_list.begin(), features_list.end()));
-  result->add_reshape("reshape", "features",
-                      {{1, num_features, 1, prediction_window}});
+
+  if (features_list.size() == 1) {
+    // Single feature column
+    const flex_string& feature_column_name = features_list.front();
+    std::string single_input_feature{feature_column_name};
+    result->add_reshape("reshape", single_input_feature,
+                        {{1, num_features, 1, prediction_window}});
+  } else {
+    // Multiple feature columns. Add concatenate layer to concat all columns.
+    result->add_channel_concat(
+        "features",
+        std::vector<std::string>(features_list.begin(), features_list.end()));
+    result->add_reshape("reshape", "features",
```

Comment:
you probably want to add similar logic for tf?

Reply:
We don't use tempfile here. Can you clarify which logic change you meant?

Reply:
@guihao-liang This has nothing to do with tf Graph or MPS Graph.

Reply:
@shreyajain17 you mean the extra concat layer won't affect the tf architecture?

Reply:
@guihao-liang  The concat layer was in the default AC graph. I added logic to not use concat layer where there isn't more than one feature columns i.e. nothing to concat.

Reply:
I understand that. this is for coreml. What I want to say is that where the `concat` logic is implemented. Is it in C++ backend before feeding the data into any compute context?

Reply:
Yes this happens in `init_model()` before instantiating the NN backend from the compute context.

Reply:
Yeah that's what I meant.

Reply:
Yes, at present the only interdependence between the CoreML spec here and what we do in the backends is the weights/parameters we use to initialize the backends. There are no NN weights or parameters involved with this change.